### PR TITLE
[prometheus-cloudwatch-exporter] Add priorityClassName to cloudwatch-exporter

### PR DESCRIPTION
**Pull request by ercpereda**
Signed-off-by: Ernesto Cruz <ernesto.cruz@alayacare.com>

### CODE REVIEW

**Comment by ercpereda** [comment]
I don't get what is the issue here (https://github.com/scottrigby/prometheus-helm-charts/pull/31/checks?check_run_id=1033262267#step:4:98)

**Comment by torstenwalter** [comment]
> I don't get what is the issue here (https://github.com/scottrigby/prometheus-helm-charts/pull/31/checks?check_run_id=1033262267#step:4:98)

That's a mistake I made when activating the linter. It should be fixed with #32. Anyhow we would need to get #1 merged first if I understand it.

**Comment by ercpereda** [comment]
> > I don't get what is the issue here (https://github.com/scottrigby/prometheus-helm-charts/pull/31/checks?check_run_id=1033262267#step:4:98)
> 
> That's a mistake I made when activating the linter. It should be fixed with #32. Anyhow we would need to get #1 merged first if I understand it.

Ok, please let me know if there is something else I can help

**Review by monotek — DISMISSED** [review]
imho adding a new feature should raise minor chart version (0.9.0).

File: charts/prometheus-cloudwatch-exporter/Chart.yaml
Comment (ercpereda) [inline charts/prometheus-cloudwatch-exporter/Chart.yaml:5]:
```suggestion
version: 0.9.0
```

**Comment by ercpereda** [comment]
I forgot about the `Signed-off-by` comment.
What should I do now?

**Comment by torstenwalter** [comment]
`git commit --amend -s
git push --force`

**Comment by ercpereda** [comment]
Hello, any updates on this?

**Comment by monotek** [comment]
Please fix yamllint errors: https://github.com/scottrigby/prometheus-helm-charts/pull/31/checks?check_run_id=1043443916

**Comment by ercpereda** [comment]
> Please fix yamllint errors: https://github.com/scottrigby/prometheus-helm-charts/pull/31/checks?check_run_id=1043443916

I'm no sure what it is wrong.

**Comment by torstenwalter** [comment]
I think https://github.com/scottrigby/prometheus-helm-charts/pull/1 needs to be merged first. It contains fixes for the linter.

**Comment by scottrigby** [comment]
@torstenwalter probably best to cherry-pick the linter fix commit(s) into a new PR so we can merge it separately, since PRs for other charts unrelated to #1 will need that.

**Comment by torstenwalter** [comment]
I'll do it tomorrow.

**Comment by torstenwalter** [comment]
@ercpereda I create a PR to correct the linter settings. Once  #33 is merge you could merge `main` into this branch. That will re-trigger the jobs and the linting errors should be gone.

Sorry for the inconvenience.

**Comment by scottrigby** [comment]
@ercpereda #33 is merged 👍  merging or rebasing this PR branch should fix the issues

**Comment by ercpereda** [comment]
After merge it, this will be deployed to the same place that before (`stable/prometheus-cloudwatch-exporter`)?

**Comment by scottrigby** [comment]
After this PR is merged, it will temporarily be available in the current github pages location (`helm repo add temp-prometheus-community https://scottrigby.github.io/prometheus-helm-charts`, `helm install temp-prometheus-community/prometheus-cloudwatch-exporter`). But once #11 is complete, you will then be able to remove the temporary repo listing and `helm repo add prometheus-community https://prometheus-community.github.io/helm-charts`). Hope that makes sense.

**Comment by scottrigby** [comment]
As maintainers of this chart, will let @gianrubio @torstenwalter or @asherf finish reviewing/approve/merge this.

https://github.com/scottrigby/prometheus-helm-charts/blob/main/.github/CODEOWNERS#L7

File: charts/prometheus-cloudwatch-exporter/values.yaml
Comment (torstenwalter) [inline charts/prometheus-cloudwatch-exporter/values.yaml:208]: Do we need this line? Should it be PriorityClassName?

Comment (ercpereda) [reply charts/prometheus-cloudwatch-exporter/values.yaml:208]: That line is referring to the `PriorityClass` resource.
Sure, I can remove it if is not clear enoght

Comment (torstenwalter) [reply charts/prometheus-cloudwatch-exporter/values.yaml:208]: I was referring to `# PriorityClass: system-cluster-critical`. It's not used in templates.

You are using `priorityClassName` in templates. That's why I asked if it should be renamed in the comment.

Comment (ercpereda) [reply charts/prometheus-cloudwatch-exporter/values.yaml]: Done 👍

**Review by torstenwalter — APPROVED** [review]

**Pull request merged**
